app/processors/cora_processor.py

```diff
@@ -0,0 +1,60 @@
+from sdc.rabbit.exceptions import QuarantinableError, RetryableError
+
+from app import settings
+from app.helpers.request_helper import remote_call, get_sequence_no, response_ok
+
+
+class CoraProcessor(object):
+
+    def __init__(self, logger, survey, ftpconn):
+        self.logger = logger
+        self.survey = survey
+        self.tx_id = None
+        self._setup_logger()
+        self.ftp = ftpconn
+
+    def process(self):
+        transformed = self._transform()
+
+        delivered = self.ftp.unzip_and_deliver(self._get_ftp_folder(self.survey), transformed)
+
+        if not delivered:
+            self.logger.error("Failed to deliver zip to ftp")
+            raise RetryableError("Failed to deliver zip to ftp")
+
+        return
+
+    def _setup_logger(self):
+        if self.survey:
+            if 'metadata' in self.survey:
+                metadata = self.survey['metadata']
+                self.logger = self.logger.bind(user_id=metadata['user_id'], ru_ref=metadata['ru_ref'])
+
+            if 'tx_id' in self.survey:
+                self.tx_id = self.survey['tx_id']
+                self.logger = self.logger.bind(tx_id=self.tx_id)
+
+    def _get_url(self):
+        sequence_no = get_sequence_no()
```

Comment:
Will this work with this pull request? https://github.com/ONSdigital/sdx-sequence/pull/43

Reply:
I spoke to Dave and this should work with the sequence changes.